src/Providers/Stripe/Card/Service.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace PodPoint\Payments\Providers\Stripe\Card;
+
+use PodPoint\Payments\Card;
+use PodPoint\Payments\Card\Service as ServiceInterface;
+use PodPoint\Payments\Providers\Stripe\Card\Exception as CardException;
+use PodPoint\Payments\Providers\Stripe\Token as StripeToken;
+use PodPoint\Payments\Token;
+use Stripe\SetupIntent;
+use Stripe\PaymentMethod;
+
+class Service implements ServiceInterface
+{
+    /**
+     * Tries create a card using the Stripe SDK.
+     *
+     * @param Token|null $token
+     *
+     * @return Card
+     *
+     * @throws CardException
+     */
+    public function create(Token $token = null): Card
+    {
+        if ($token) {
+            /** @var SetupIntent $response */
+            $response = SetupIntent::retrieve($token->value);
+        } else {
+            $params = [
```

Comment:
I don't see benefit for it to be held in a variable.

Reply:
Thought it was better for readability but I removed it ;-)